services/data-info/src/data_info/routes/avus.clj

```diff
@@ -12,26 +13,29 @@
 
     (GET* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :return AVUGetResult
+      :return MetadataListing
+      :middlewares [wrap-metadata-base-url]
       :summary "List AVUs (administrative)"
-      :description (str "List iRODS AVUs associated with a data item. Include administrative/system AVUs."
-(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE"))
+      :description
+          (str "List all AVUs associated with a data item. Include administrative/system iRODS AVUs."
+(get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE")
+(get-endpoint-delegate-block "metadata" "GET /avus/{target-type}/{target-id}"))
       (svc/trap uri meta/admin-metadata-get data-id))
 
-    (POST* "/avus" [:as {uri :uri}]
+    (PUT* "/avus" [:as {uri :uri}]
       :query [{:keys [user]} StandardUserQueryParams]
-      :body [{:keys [irods-avus]} (describe AVUListing "An list of AVUs to add")]
+      :body [{:keys [irods-avus]} (describe AVUListing "A list of AVUs to add")]
       :return AVUChangeResult
       :summary "Add AVUs (administrative)"
-      :description (str "Associate AVUs with a data item. Allow adding any AVU."
+      :description (str "Associate iRODS AVUs with a data item. Allow adding any AVU."
 (get-error-code-block "ERR_NOT_A_USER, ERR_NOT_READABLE, ERR_DOES_NOT_EXIST, ERR_NOT_WRITEABLE, ERR_NOT_AUTHORIZED"))
       (svc/trap uri meta/admin-metadata-add data-id irods-avus))
```

Comment:
I don't have strong feelings about PATCH, ultimately. I don't like it because there doesn't seem to be much of a consensus, so I tend to be conservative and use POST instead, but I wouldn't be opposed to using it either (my complaints are more aesthetic than practical). The point about deletes was mostly just because that's how I tend to think about PATCH, but nobody seems to really agree on PATCH, so it's not a requirement, just a feeling! Sorry I wasn't that clear about that.

Mostly, the only thing that is clearly wrong to me is using PUT for something which does not send an entire resource representation (the whole list) at once, which I think you agree on.

As for the rest: since we want the same URI, they can't both be POST. I think PUT is clear enough/idempotent enough to be used for the set endpoint (which is what most things in our code, and externally from what I've seen, use PUT for). POST would also be basically "fine" for the set endpoint, but it's sort of a cop-out on using HTTP verbs to denote semantics. I don't have a strong opinion about whether the add/update should be POST or PATCH, though I'd choose POST by the conservatism arguments I list above and because I prefer PUT for the set endpoint it's available.

Basically, I put these into data-info as PUT for set and POST for add+update because that's my preference, but the only arrangement that I see as definitely wrong is using PUT for add+update. set/add as PUT/POST, POST/PATCH, and PUT/PATCH all seem, if not perfect, at least not wrong. PUT/PATCH is probably the most correct, where PUT/POST is the most if you exclude PATCH due to ill-definedness like I tend to.